src/Compilers/Core/Portable/SourceGeneration/WellKnownGeneratorInputs.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Well known incremental generator input step names.
+    /// </summary>
+    public static class WellKnownGeneratorInputs
+    {
+        public const string Compilation = nameof(Compilation);
+
+        public const string ParseOptions = nameof(ParseOptions);
+
+        public const string AdditionalTexts = nameof(AdditionalTexts);
+
+        public const string SyntaxTrees = nameof(SyntaxTrees);
```

Comment:
I'm surprised that we expose this? It's really an implementation detail of how the syntax nodes work. I don't quite understand how it would be used by the end user.

Reply:
When I initially implemented the input tracking, I asked @sharwell if we should expose the `SyntaxTree`s node, and he said that we should expose graph all the way back up to the SyntaxTree.

Reply:
I still think we shouldn't be exposing this. As a user you never explicitly chain off of a `SyntaxTree` value provider. Instead you create your own value provider via `SyntaxValueProvider.CreateSyntaxProvider` so the root node of any syntax based operations should be that node itself. 

You shouldn't be able to see from the tests 'oh this syntax tree didn't change etc, it should just be 'These are the syntax nodes we picked (or were cached), here are the transformations we performed'